test/e2e/common_test.go

```diff
@@ -404,15 +426,155 @@ func PodmanTestCreateUtil(tempDir string, remote bool) *PodmanTestIntegration {
 				lockFile.Close()
 				p.RemoteSocketLock = lockPath
 				p.RemoteSocket = fmt.Sprintf("unix://%s-%s.sock", pathPrefix, uuid)
+				p.RemoteSocketScheme = "unix"
 				break
 			}
+			GinkgoLogr.Error(err, "RemoteSocket collision")
+			tries++
+			if tries >= 1000 {
+				panic("Too many RemoteSocket collisions")
+			}
+		}
+	case PodmanTestCreateUtilTargetTCP, PodmanTestCreateUtilTargetTLS, PodmanTestCreateUtilTargetMTLS:
+		tries := 0
+		for {
+			uuid := stringid.GenerateRandomID()
+			lockPath := fmt.Sprintf("%s-%s.sock-lock", pathPrefix, uuid)
+			lockFile, err := os.OpenFile(lockPath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0700)
+			if err == nil {
+				lockFile.Close()
+				p.RemoteSocketLock = lockPath
+				lis, err := net.Listen("tcp", "127.0.0.1:0")
+				if err == nil {
+					defer lis.Close()
+					p.RemoteSocket = fmt.Sprintf("tcp://%s", lis.Addr())
```

Comment:
this is not race safe, if we do this for every single test I can guarantee this will flake quite often.
the port is freed before the service can bind it. You would need to pick ports not from the dynamic range and lock them so other tests don't use it in parallel. Or maybe better we make use of the socket activation feature instead. We can pass the fd to the podman service by using the systemd socket activation protocol.